packages/tools/viewer/src/viewerFactory.ts

```diff
@@ -16,7 +16,15 @@ const defaultCanvasViewerOptions: CanvasViewerOptions = {};
  * @returns The default engine to use.
  */
 export function getDefaultEngine(): NonNullable<CanvasViewerOptions["engine"]> {
-    // TODO: When WebGPU is fully production ready, we may want to prefer it if it is supported by the browser.
+    // First check for WebGPU support.
+    if ("gpu" in navigator) {
+        // For now, only use WebGPU with chromium-based browsers.
+        // WebGPU can be enabled in other browsers once they are fully functional and the performance is at least as good as WebGL.
+        if ("chrome" in window) {

```

Comment:
I remember seeing a discussion about whether or not edge will keep this variable. Just checked and it is there, so I guess it is safe, but is there no better way to detect a full support?

Reply:
I believe we could also detect it from the userAgent, it's just more parsing code. I'm not sure which way is better.

Reply:
UA sniffing is always a bad idea. I was just wondering if there is any feature-detection we can do. Like check for some specific WebGPU-related class that exists in chrome but not other browsers. But if others have implemented the specs just simply not well, there is probably nothing we can do.

Reply:
`window.chrome` probably only exists in Chrome/Edge?

Reply:
Yes, but there is no reason it will forever stay in edge. And no was guarantees that firefox will not decide to implement it for whatever reason. Gunny enough, old edge browsers had it defined. Probably to to avoid chrome detection. 
My point is that we should be running feature-detection and not detect browsers using these kind of methods.
Feel free to resolve this, if there is no other solution for this.

Reply:
> But if others have implemented the specs just simply not well, there is probably nothing we can do.
This is really the main issue. For example, in Safari, WebGPU can be enabled and the API is there, but performance is much worse the WebGL so we really wouldn't want to default to WebGPU in this case. We want users to have the best experience possible by default.